test/addons/rook-external-cluster/start

```diff
@@ -0,0 +1,176 @@
+    script_name = ''
+
+    _, script_name = tempfile.mkstemp(prefix='import-external-cluster-', suffix='.sh')
+
+    download_script_result = subp.run(['curl', '-s', 'https://raw.githubusercontent.com/raghavendra-talur/rook/rtalur-support-context/deploy/examples/import-external-cluster.sh', '-o', script_name], capture_output=True)
+    # download_script_result = subp.run(['curl', '-s', 'https://raw.githubusercontent.com/rook/rook/master/deploy/examples/import-external-cluster.sh', '-o', script_name], capture_output=True)
+    print(f"script {script_name}")
+
+    if download_script_result.returncode != 0:
+        print("Failed to download script: " + download_script_result.stderr)
+        sys.exit(1)
+
+    if not os.path.isfile(script_name):
+        print("Script file does not exist: " + script_name)
+        sys.exit(1)
+
+    with open(script_name) as sf:
+        src = sf.read()
+
+    src = src.replace("RBD_STORAGE_CLASS_NAME=ceph-rbd", "RBD_STORAGE_CLASS_NAME=rook-ceph-block")
+    src = src.replace("CEPHFS_STORAGE_CLASS_NAME=cephfs", "CEPHFS_STORAGE_CLASS_NAME=rook-ceph-fs")
+
+    if not os.access(script_name, os.W_OK):
+        print("Cannot write to script file: " + script_name)
+        sys.exit(1)
+
+    with open(script_name, 'w') as sf:
+        sf.write(src)
+
+    env['NAMESPACE'] = ROOK_CEPH_NAME_SPACE
+    env['KUBECONTEXT'] = managed_cluster
+
+    for line in cluster_details.split('\n'):
+
+        if line.startswith("export "):
+            _, rest = line.split("export ", 1)
+            k, v = rest.split("=", 1)
+            env[k] = v
+
+        #if line.startswith('export'):
+        #    _, v = line.split(' ')
+        #    bits = v.split('=')
+        #    k = bits[0]
+        #    v = '='.join(bits[1:])
+        #    env[k] = v
+
+
+    print(f"env {env}")
+    kubectl.config('use-context', managed_cluster)
+    print(f"context {kubectl.config('current-context')}")
+    run_script_result = subp.run(shlex.split(' '.join(['/bin/bash', script_name])), capture_output=True, encoding='utf-8', env=env)
+
+    if run_script_result.returncode != 0:
+        print("Failed to run script: " + run_script_result.stderr)
+        sys.exit(1)
+
+    print("Script output: " + run_script_result.stdout)
+    sys.exit(0)
+
+    os.unlink(script_name)
+
+def get_ceph_toolbox_pod_name(cluster):
+    return kubectl.get(
+        "pod",
+        "-l app=rook-ceph-tools",
+        "--output=jsonpath={.items[0].metadata.name}",
+        f"--namespace={ROOK_CEPH_NAME_SPACE}",
+        context=cluster
+    )
+
+def get_rook_ceph_external_cluster_details(podname, cluster):
+    try:
+        out = kubectl.exec(
+            podname,
+            f"--namespace={ROOK_CEPH_NAME_SPACE}",
+            "--",
+            "curl",
+            "-o",
+            "/tmp/create-external-cluster-resources.py",
+            "https://raw.githubusercontent.com/rook/rook/master/deploy/examples/create-external-cluster-resources.py",
```

Comment:
Running code from github blindly is not something developer should accept.
We better copy this code into our code so we have more control.